llvm/lib/Transforms/Scalar/SROA.cpp

```diff
@@ -627,11 +627,8 @@ class AllocaSlices {
   /// everything so that the usual ordering properties of the alloca's slices
   /// hold.
   void insert(ArrayRef<Slice> NewSlices) {
-    int OldSize = Slices.size();
     Slices.append(NewSlices.begin(), NewSlices.end());
-    auto SliceI = Slices.begin() + OldSize;
-    llvm::sort(SliceI, Slices.end());
-    std::inplace_merge(Slices.begin(), SliceI, Slices.end());
```

Comment:
Why did this switch to sorting the whole Slices? Isn't inplace_merge stable?

Reply:
At first I tried to replace `llvm::sort` with `llvm::stable_sort` but that resulted in a bunch of template errors. Since this function seemed to be extremely cold, I went with this simplification instead. During a full build of clang+lld cross-compiling themselves, this function was only called 3 times in total.

Thinking about this again, I realise that I should probably have replaced `llvm::sort` with `std::stable_sort` (not `llvm::stable_sort`). Is that right?

Reply:
Yeah, that's right. `llvm::stable_sort` is just a thin wrapper around `std::stable_sort` that calls begin() and end() on the container. When passing explicit iterators you use `std::stable_sort`.

(`llvm::sort` is different, because it also has a special POD sort implementation.)